core/blockchain.go

```diff
@@ -1048,11 +1088,9 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
 			batch.Reset()
 		}
 	}
-	if batch.ValueSize() > 0 {
```

Comment:
why remove the > 0 condition?

Reply:
This is to make `InsertReceiptChain` atomic. Else there will be some unwritten data at the end of `InsertReceiptChain` (though the function is not used in production code now)

Reply:
Ummm, there is no strong reason to do this change. Let me switch it back.